python_modules/dagster/dagster/_core/host_representation/external_data.py

```diff
@@ -588,6 +588,7 @@ def __new__(
                 target_dict, "target_dict", str, ExternalTargetData
             ),
             metadata=check.opt_inst_param(metadata, "metadata", ExternalSensorMetadata),
+            # Leave default_status as None if it's STOPPED to maintain stable back-compat IDs
```

Comment:
There's a similar backcompat breadcrumb on `ExternalScheduleData`, so thought I'd leave this here.

Reply:
I don't think I quite understand how this comment gibes with switching the type to always return a `DefaultSensorStatus` (e.g. not handling the `None` type)

Reply:
It was pretty confusing for me as well, but here's my understanding:

- For backcompat reasons, the DefaultXXXStatus can be None. This handles the case when default status didn't exist in the API.
- We continue to allow default status as None in storage because we don't want to change the existing ids/hashes of the serialized objects.
- When we deserialize the object, we want to properly interpret these None's. These None's should be interpreted as the default argument for the default_status parameter in @sensor/@schedule, which is DefaultXXXStatus.STOPPED. 
- So even though the storage object representation of default_status is optional, we always know what it is (either it is what we have in storage, or we default to DefaultXXXStatus.STOPPED)

Reply:
Got it... so the external sensor/schedule will always return a status but the data object might have status None... and that's the logic the `ExternalSchedule` already had.

Thanks